hw/top_earlgrey/syn/asic.constraints.sdc

```diff
@@ -4,205 +4,180 @@
+#####################
+set SPI_CLK_PIN SPI_DEV_CLK
+# 62.5MHz
+set SPI_TCK 16.0
+set_ideal_network ${SPI_CLK_PIN}
+
+create_clock -name SPID_CLK  -period ${SPI_TCK} [get_ports ${SPI_CLK_PIN}]
+set_clock_uncertainty ${SETUP_CLOCK_UNCERTAINTY} [get_clocks SPID_CLK]
+
+## TODO: Create generated clock for negedge SPID_CLK. Then make them clock group
 
-set IN_DEL    10.0
-set OUT_DEL   10.0
+## TODO: these are dummy constraints and likely incorrect, need to properly constrain min/max
+set IN_DEL    6.0
+set OUT_DEL   6.0
 
-set JTAG_TMS_PIN IO_DPS3
-set JTAG_TDI_PIN IO_DPS1
-set JTAG_TDO_PIN IO_DPS2
-set_input_delay ${IN_DEL} [get_ports IO_DPS3]  -clock JTAG_CLK
-set_input_delay ${IN_DEL} [get_ports IO_DPS1]   -clock JTAG_CLK
+# this is an input only port
+set_input_delay ${IN_DEL} [get_ports SPI_DEV_CS_L] -clock SPID_CLK
+# bidir ports
+set_input_delay ${IN_DEL} [get_ports SPI_DEV_D0]   -clock SPID_CLK
+set_input_delay ${IN_DEL} [get_ports SPI_DEV_D1]   -clock SPID_CLK
+set_input_delay ${IN_DEL} [get_ports SPI_DEV_D2]   -clock SPID_CLK
+set_input_delay ${IN_DEL} [get_ports SPI_DEV_D3]   -clock SPID_CLK
 
-set_output_delay ${OUT_DEL} [get_ports IO_DPS2] -clock JTAG_CLK
+set_output_delay ${OUT_DEL} [get_ports SPI_DEV_D0]   -clock SPID_CLK
+set_output_delay ${OUT_DEL} [get_ports SPI_DEV_D1]   -clock SPID_CLK
+set_output_delay ${OUT_DEL} [get_ports SPI_DEV_D2]   -clock SPID_CLK
+set_output_delay ${OUT_DEL} [get_ports SPI_DEV_D3]   -clock SPID_CLK
 
 #####################
-# SPI clock         #
+# SPI HOST clock   #
 #####################
-set SPI_CLK_PIN dio_in_core[$PORT_SPI_DEVICE_SCK]
+set SPI_CLK_PIN SPI_HOST_CLK
 # 62.5MHz
 set SPI_TCK 16.0
 set_ideal_network ${SPI_CLK_PIN}
 
-create_clock -name SPID_CLK  -period ${SPI_TCK} [get_pins ${SPI_CLK_PIN}]
-set_clock_uncertainty ${SETUP_CLOCK_UNCERTAINTY} [get_clocks SPID_CLK]
+create_clock -name SPIH_CLK  -period ${SPI_TCK} [get_ports ${SPI_CLK_PIN}]
+set_clock_uncertainty ${SETUP_CLOCK_UNCERTAINTY} [get_clocks SPIH_CLK]
 
-## TODO: Create generated clock for negedge SPID_CLK. Then make them clock group
+## TODO: Create generated clock for negedge SPIH_CLK. Then make them clock group
 
+## TODO: these are dummy constraints and likely incorrect, need to properly constrain min/max
 set IN_DEL    6.0
 set OUT_DEL   6.0
 
-set_input_delay ${IN_DEL} [get_ports dio_in_core[$PORT_SPI_DEVICE_CSB]]   -clock SPID_CLK
-set_input_delay ${IN_DEL} [get_ports dio_in_core[$PORT_SPI_DEVICE_S0]]    -clock SPID_CLK
-set_input_delay ${IN_DEL} [get_ports dio_in_core[$PORT_SPI_DEVICE_S1]]    -clock SPID_CLK
-set_input_delay ${IN_DEL} [get_ports dio_in_core[$PORT_SPI_DEVICE_S2]]    -clock SPID_CLK
-set_input_delay ${IN_DEL} [get_ports dio_in_core[$PORT_SPI_DEVICE_S3]]    -clock SPID_CLK
+# bidir ports
+set_input_delay ${IN_DEL} [get_ports SPI_HOST_CS_L] -clock SPIH_CLK
+set_input_delay ${IN_DEL} [get_ports SPI_HOST_D0]   -clock SPIH_CLK
+set_input_delay ${IN_DEL} [get_ports SPI_HOST_D1]   -clock SPIH_CLK
```

Comment:
no need to change anything for now, but let me double check long term.
There may not be a command use case for quad commands -> out, and only quad data in.

If that's true, we probably don't need to set output_delay on D2/D3, maybe even D1.